packages/angular_devkit/build_angular/src/builders/karma/index.ts

```diff
@@ -110,45 +106,22 @@ export function execute(
         }
       }
 
-      // prepend special webpack loader that will transform test.ts
-      if (options.include?.length) {
-        const projectName = context.target?.project;
-        if (!projectName) {
-          throw new Error('The builder requires a target.');
-        }
+      const projectName = context.target?.project;
+      if (!projectName) {
+        throw new Error('The builder requires a target.');
+      }
 
-        const projectMetadata = await context.getProjectMetadata(projectName);
-        const sourceRoot = (projectMetadata.sourceRoot ?? projectMetadata.root ?? '') as string;
-        const projectSourceRoot = path.join(context.workspaceRoot, sourceRoot);
+      const projectMetadata = await context.getProjectMetadata(projectName);
+      const sourceRoot = (projectMetadata.sourceRoot ?? projectMetadata.root ?? '') as string;
 
-        const files = await findTests(options.include, context.workspaceRoot, projectSourceRoot);
-        // early exit, no reason to start karma
-        if (!files.length) {
-          throw new Error(
-            `Specified patterns: "${options.include.join(', ')}" did not match any spec files.`,
-          );
-        }
-
-        // Get the rules and ensure the Webpack configuration is setup properly
-        const rules = webpackConfig.module?.rules || [];
-        if (!webpackConfig.module) {
-          webpackConfig.module = { rules };
-        } else if (!webpackConfig.module.rules) {
-          webpackConfig.module.rules = rules;
-        }
-
-        rules.unshift({
-          test: path.resolve(context.workspaceRoot, options.main),
-          use: {
-            // cannot be a simple path as it differs between environments
-            loader: SingleTestTransformLoader,
-            options: {
-              files,
-              logger: context.logger,
-            },
-          },
-        });
-      }
+      webpackConfig.plugins ??= [];
+      webpackConfig.plugins?.push(
```

Comment:
nit: `plugins` should always be defined via the previous statement

Reply:
You are right 👍